controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -780,8 +806,11 @@ func (r *BareMetalHostReconciler) manageHostPower(prov provisioner.Provisioner,
 	info.log.Info("power state change needed",
 		"expected", desiredPowerOnState,
 		"actual", info.host.Status.PoweredOn,
+		"hard reboot", desiredHardReboot,
 		"reboot process", desiredPowerOnState != info.host.Spec.Online)
 
+	// WIP (rdo) The isHardReboot functionality is a WIP as the provisioner
+	// API needs changes to allow us to call hardPowerOff here if requested
```

Comment:
I think you'll need to switch the order of the commits, then you can remove this and pass `desiredHardReboot` into `prov.PowerOff()` below?

Reply:
That's correct. The other commit listed in the PR adds the desiredHardReboot into the PowerOff()